Packs/ArmisEventCollector/Integrations/ArmisEventCollector/ArmisEventCollector.py

```diff
@@ -0,0 +1,427 @@
+        send_events_to_xsiam(
+            events,
+            vendor=VENDOR,
+            product=PRODUCT
+        )
+        demisto.setLastRun(next_run)
+        demisto.debug(f'debug-log: {len(events)} events were sent to XSIAM.')
+        demisto.debug(f'debug-log: {next_run=}')
+    else:
+        demisto.debug('debug-log: No new events fetched, Last run was not updated.')
+
+
+def events_to_command_results(events: list[dict[str, Any]]) -> CommandResults:
+    """ Return a CommandResults object with a table of fetched events.
+
+    Args:
+        events (list[dict[str, Any]]): list of fetched events.
+
+    Returns:
+        CommandResults: CommandResults object with a table of fetched events.
+    """
+    return CommandResults(
+        raw_response=events,
+        readable_output=tableToMarkdown(name=f'{VENDOR} {PRODUCT} events',
+                                        t=events,
+                                        removeNull=True))
+
+
+''' MAIN FUNCTION '''
+
+
+def main():
+    params = demisto.params()
+    args = demisto.args()
+    command = demisto.command()
+    last_run = demisto.getLastRun()
+    access_token = last_run.get('access_token')
+    api_key = params.get('credentials', {}).get('password')
+    base_url = urljoin(params.get('server_url'), API_V1_ENDPOINT)
+    verify_certificate = not params.get('insecure', True)
+    max_fetch = arg_to_number(params.get('max_fetch')) or DEFAULT_MAX_FETCH
+    proxy = params.get('proxy', False)
+    event_types_to_fetch = argToList(params.get('event_types_to_fetch', []))
+    should_push_events = argToBoolean(args.get('should_push_events', False))
+    from_date_datetime = None
+    if from_date := args.get('from_date'):
+        from_date_datetime = handle_from_date_argument(from_date)
+
+    demisto.debug(f'Command being called is {command}')
+
+    try:
+        client = Client(
+            base_url=base_url,
+            verify=verify_certificate,
+            proxy=proxy,
+            api_key=api_key,
+            access_token=access_token)
+
+        if command == 'test-module':
+            return_results(test_module(client))
+
+        elif command in ('fetch-events', 'armis-get-events'):
+            push_events = (command == 'fetch-events' or should_push_events)
+
+            events, next_run = fetch_events(
+                client=client,
+                max_fetch=max_fetch,
+                last_run=last_run,
+                fetch_start_time_param=from_date_datetime or datetime.now(),
+                event_types_to_fetch=event_types_to_fetch,
+            )
+
+            demisto.debug(f'debug-log: {len(events)} events fetched from armis api')
+
+            if push_events:
+                handle_fetched_events(events, next_run)
+
+            if command == 'armis-get-events':
+                return_results(events_to_command_results(events))
+
+        else:
+            return_error(f'Command {command} does not exist for this integration.')
```

Comment:
Why not use raising NotImplementedError?

Reply:
Fixed.